internal/federationout/federationout.go

```diff
@@ -86,153 +86,163 @@ func (s Server) Fetch(ctx context.Context, req *pb.FederationFetchRequest) (*pb.
+			IntervalCount:  exp.IntervalCount,
 		}
 
-		// If all the regions on the record are excluded, skip it.
-		skip := true
-		for _, region := range inf.Regions {
-			if _, excluded := excludedRegions[region]; !excluded {
-				// At least one region for the exposure is NOT excluded, so we don't skip this record.
-				skip = false
-				break
+		if !request.revised {
+			// primary keys first
+			key.TransmissionRisk = int32(exp.TransmissionRisk)
+			key.ReportType = reportType(exp.ReportType)
+			if exp.HasDaysSinceSymptomOnset() {
+				key.DaysSinceOnsetOfSymptoms = *exp.DaysSinceSymptomOnset
 			}
-		}
-		if skip {
-			logger.Debugf("Exposure %s contains only excluded regions, skipping.", inf.ExposureKey)
-			return nil
-		}
 
-		// If filtering on a region (len(includedRegions) > 0) and none of the regions on the record are included, skip it.
-		if len(includedRegions) > 0 {
-			skip = true
-			for _, region := range inf.Regions {
-				if _, included := includedRegions[region]; included {
-					skip = false
-					break
-				}
+			created := exp.CreatedAt
+			if ts := created.Unix(); ts > request.state.KeyCursor.Timestamp {
+				request.state.KeyCursor.Timestamp = ts
 			}
-			if skip {
-				logger.Debugf("Exposure %s does not contain requested regions, skipping.", inf.ExposureKey)
-				return nil
+		} else {
+			// Revised keys get different fields
+			key.TransmissionRisk = int32(*exp.RevisedTransmissionRisk)
+			key.ReportType = reportType(*exp.RevisedReportType)
+			if exp.RevisedDaysSinceSymptomOnset != nil {
+				key.DaysSinceOnsetOfSymptoms = *exp.RevisedDaysSinceSymptomOnset
 			}
-		}
-
-		// Find, or create, the ContactTracingResponse based on the unique set of regions.
-		sort.Strings(inf.Regions)
-		ctrKey := strings.Join(inf.Regions, "::")
-		ctr := ctrMap[ctrKey]
-		if ctr == nil {
-			ctr = &pb.ContactTracingResponse{RegionIdentifiers: inf.Regions}
-			ctrMap[ctrKey] = ctr
-			response.Response = append(response.Response, ctr)
-		}
 
-		// Find, or create, the ContactTracingInfo for (ctrKey, transmissionRisk).
-		ctiKey := fmt.Sprintf("%s::%d", ctrKey, inf.TransmissionRisk)
-		cti := ctiMap[ctiKey]
-		if cti == nil {
-			cti = &pb.ContactTracingInfo{TransmissionRisk: int32(inf.TransmissionRisk)}
-			ctiMap[ctiKey] = cti
-			ctr.ContactTracingInfo = append(ctr.ContactTracingInfo, cti)
+			revisedAt := *exp.RevisedAt
+			if ts := revisedAt.Unix(); ts > request.state.RevisedKeyCursor.Timestamp {
+				request.state.RevisedKeyCursor.Timestamp = ts
+			}
 		}
 
-		// Add the key to the ContactTracingInfo.
-		cti.ExposureKeys = append(cti.ExposureKeys, &pb.ExposureKey{
-			ExposureKey:    inf.ExposureKey,
-			IntervalNumber: inf.IntervalNumber,
-			IntervalCount:  inf.IntervalCount,
-		})
-
-		created := inf.CreatedAt.Unix()
-		if created > response.FetchResponseKeyTimestamp {
-			response.FetchResponseKeyTimestamp = created
-		}
+		request.destination = append(request.destination, &key)
 
-		count++
+		*request.count++
```

Comment:
I think this will panic if count is nil? Is that possible?

Reply:
it's always initialized in the current code.